### PR TITLE
API metrics optional

### PR DESCRIPTION
**Pull request by HvonP**
**Checklist**

- [ ] tests added
- [x] commits conform to the [Commit message format](https://github.com/qlik-oss/gopherciser/blob/master/.github/CONTRIBUTING.md#commit)
- [x] documentation updated

**Squash commit message**

API metrics optional with reworked metrics arguments

**Info**

Replaced metricsport with metricslevel and metricsaddress with metricstarget. This allows api metrics to be turned on or off at will

### CODE REVIEW

File: cmd/execute.go [hunk 1]
```diff
@@ -49,8 +49,8 @@ type (
 )
 
 var (
-	metricsPort      int
-	metricsAddress   string
```
Comment (DnlLrssn) [inline cmd/execute.go:53]: should probably keep and deprecate these, if set, convert them to the new settings and print a warning they are deprectated. Also simplifies rolling out this breaking change.

File: buildmetrics/buildmetrics.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build linux || darwin || windows
```
Comment (atluq) [inline buildmetrics/buildmetrics.go:1]: what side effect do this have?

**Review by atluq — COMMENTED** [review]
Agree with previous comments

File: cmd/execute.go [hunk 3]
```diff
@@ -305,21 +305,29 @@ func execute() error {
...
-			if err != nil {
-				return MetricError(fmt.Sprintf("failed to start prometheus : %s ", err))
-			}
-		} else {
-			//Prometheus metrics will be pushed to a pushgateway
-			err := buildmetrics.PushMetrics(ctx, metricsPort, metricsAddress, metricsLabel, metricsGroupings, scenario.RegisteredActions())
-			if err != nil {
-				return MetricError(fmt.Sprintf("failed to start prometheus : %s ", err))
-			}
+	switch metricsLevel {
```
[the remark this answers is not in the document]
Comment (HvonP) [reply cmd/execute.go]: Yeah done

File: cmd/execute.go (hunk 1 above)
Comment (HvonP) [reply cmd/execute.go]: Good idea. Added a crude parser that automatically turns off API level metrics

File: cmd/execute.go [hunk 4]
```diff
@@ -305,21 +351,62 @@ func execute() error {
...
-			//Prometheus metrics will be pulled from the endpoint /metrics
-			err := buildmetrics.PullMetrics(ctx, metricsPort, scenario.RegisteredActions())
-			if err != nil {
-				return MetricError(fmt.Sprintf("failed to start prometheus : %s ", err))
+	if metricsLevel != "" || metricsPort > 0 {
+		var metricsType MetricLevel
+		var err error
+
+		// Temporary conversion from legacy code metrics arguments to new
+		if metricsPort > 0 {
```
Comment (DnlLrssn) [inline cmd/execute.go:359]: 👍 would be good to also print a warning to console for usage of deprecated parameter so the user knows.

File: cmd/execute.go [hunk 5]
```diff
@@ -213,8 +257,10 @@ func init() {
 	AddLoggingParameters(executeCmd)
 
 	// Prometheus
-	executeCmd.Flags().IntVar(&metricsPort, "metrics", 0, "Export via http prometheus metrics.")
-	executeCmd.Flags().StringVar(&metricsAddress, "metricsaddress", "", "If set other than empty string then Push otherwise pull, will be appended by port.")
+	executeCmd.Flags().StringVar(&metricsLevel, "metricslevel", "", "Export via http prometheus metrics. 0-off, 1-Pull, 2-Push without api, 3 push with api")
```
Comment (DnlLrssn) [inline cmd/execute.go:260]: update to reflect string names of enum, i.e. maybe something like
..."\n\t[0] nometrics (default)\n\t[1] pullmetric"
or similar so user can know something more understandable then the number can be written. 
it's ok write string like that, if you want to do it more advanced you can generate this help from the map like we do with loglevel

Comment (HvonP) [reply cmd/execute.go]: Good suggestion, added

File: buildmetrics/buildmetrics.go (hunk 2 above)
Comment (HvonP) [reply buildmetrics/buildmetrics.go:1]: It's that it only builds for those targets
For other targets there is a corresponding side implementation

Almost like an hardware/target interface

In this case there is a _JS file that nil implements JS bits

Comment (atluq) [reply buildmetrics/buildmetrics.go:1]: Cool, thanks!

**Pull request merged**